RELEASE.md

```diff
@@ -1,12 +1,14 @@
 # Jaeger Backend Release Process
 
-1. Create a PR "Preparing release X.Y.Z" against master branch ([example](https://github.com/jaegertracing/jaeger/pull/543/files)) by updating CHANGELOG.md to include:
+1. Create a PR "Preparing release X.Y.Z" against master or maintenance branch ([example](https://github.com/jaegertracing/jaeger/pull/543/files)) by updating CHANGELOG.md to include:
     * A new section with the header `<X.Y.Z> (YYYY-MM-DD)`
     * A curated list of notable changes and links to PRs. Do not simply dump git log, select the changes that affect the users.
     * The section can be split into sub-section if necessary, e.g. UI Changes, Backend Changes, Bug Fixes, etc.
 2. After the PR is merged, create a release on Github:
     * Title "Release X.Y.Z" 
-    * Tag `vX.Y.Z` (note the `v` prefix)
+    * Tag `vX.Y.Z` (note the `v` prefix) and choose appropriate branch
     * Copy the new CHANGELOG.md section into the release notes
 3. The release tag will trigger a build of the docker images
 4. Once the images are available on [Docker Hub](https://hub.docker.com/r/jaegertracing/), announce the release on the mailing list, gitter, and twitter.
+
+Maintenance should follow naming convention: `release-major.minor`.
```

Comment:
Without the `v` prefix, I assume.

Talking about branches: it would probably be a good time to clean up the branches in the main repo. None of them seem to belong here, as they look like feature branches that would originate PRs. As such, they belong to personal forks, IMO. The only exception is the branch `release-1.8`, which also seems misplaced.

Reply:
yes without `v` prefix. The v prefix is specified explicitly in this doc so major and minor represents numbers.

I agree about the branches I have added it to the meeting agenda but we can discuss it here. I think here should be only maintenance branches or branches where multiple people collaborate. We can use feature branches in forks like in other repos.

cc @jaegertracing/jaeger-maintainers

Reply:
> The only exception is the branch release-1.8, which also seems misplaced.

@jpkrohling Not sure what you mean "misplaced"?

Reply:
@pavolloffay Possibly need to give an example at the end of this line, just for clarity, e.g. "release-major.minor, e.g. `release-1.8`."

Reply:
Sorry, I didn't realize that it was *the* 1.8 branch: I thought it was a tag that got erroneously created as a branch. Looking closely to the diff in this PR, I see that the `release` is indeed part of the branch.

`release-major.minor`

I would really suggest removing the `release-` part, just keeping the version, plus a `.x` at the end:

> `{major}.{minor}.x`, like, 1.8.x

Reply:
The convention was selected, as it is used by Kubernetes and Istio.

Reply:
I guess I'm just outdated then :)

Reply:
:+1: I will add an example